### PR TITLE
 Adds fixer-break-line-after option

### PR DESCRIPTION
**Pull request by mzyil**
Add an option to enforce a new line after jsdoc or decorator tag #19

### CODE REVIEW

File: rules/explicitOverrideRule.ts [hunk 1]
```diff
@@ -57,12 +59,13 @@ export class Rule extends Lint.Rules.TypedRule {
             * \`"${OPTION_JSDOC_TAG}"\` (default) Uses a jsdoc tag: \`/** @override */ method() { }\`
             * \`"${OPTION_EXCLUDE_INTERFACES}"\` Exclude interfaces from member override checks (default: false)
             * \`"${OPTION_FIX_PASCAL_CASE}"\` Uses PascalCase \`@Override\` for the jsdoc tag or decorator in the fixer (default: false)
+            * \`"${OPTION_BREAK_LINE_AFTER}"\` Breaks the line after jsdoc tag or decorator in the fixer (default: false)
```
Comment (thehcma) [inline rules/explicitOverrideRule.ts]: after _the_ jsdoc

Comment (mzyil) [reply rules/explicitOverrideRule.ts]: Fixed

**Comment by mzyil** [comment]
@hmil ping

File: rules/explicitOverrideRule.ts [hunk 2]
```diff
@@ -57,12 +59,13 @@ export class Rule extends Lint.Rules.TypedRule {
...
         `,
         options: {
             type: 'array',
             items: {
                 type: 'string',
-                enum: [OPTION_DECORATOR, OPTION_JSDOC_TAG, OPTION_EXCLUDE_INTERFACES, OPTION_FIX_PASCAL_CASE],
+                enum: [OPTION_DECORATOR, OPTION_JSDOC_TAG, OPTION_EXCLUDE_INTERFACES, OPTION_FIX_PASCAL_CASE, OPTION_BREAK_LINE_AFTER],
```
Comment (thehcma) [inline rules/explicitOverrideRule.ts]: isn't "new line after decorator" a better name for this?

if yes, please replace everywhere

Comment (mzyil) [reply rules/explicitOverrideRule.ts]: it's not always the decorator, the fixer breaks after jsdoc tags too

Comment (thehcma) [reply rules/explicitOverrideRule.ts]: then "new-line-after-decorators-and-tags"

Comment (mzyil) [reply rules/explicitOverrideRule.ts]: Should I change `OPTION_BREAK_LINE_AFTER` to `OPTION_NEW_LINE_AFTER` also?

Comment (thehcma) [reply rules/explicitOverrideRule.ts]: yeah, make the whole thing consistent

File: rules/explicitOverrideRule.ts [hunk 3]
```diff
@@ -177,6 +182,8 @@ class Walker extends Lint.AbstractWalker<IOptions> {
             // But no base symbol was found
             (base === undefined || base.baseClass === undefined && base.baseInterface === undefined)
         ) {
+            // TODO: When given two @override decorators following each other (with a new line or a whitespace between them),
+            // the fixer does not break the line
```
Comment (hmil) [inline rules/explicitOverrideRule.ts:193]: What do you mean exactly? That doesn't sound like something we should worry about.

**Comment by hmil** [comment]
Sweet. That's a really clean patch, I appreciate it. Could you either address or remove the TODO?

File: rules/explicitOverrideRule.ts (hunk 3 above)
Comment (mzyil) [reply rules/explicitOverrideRule.ts:193]: Please take a look at the test files of both decorator and jsdoc tags.
```javascript
@override
@override function bar2(): void { };
```
becomes
```javascript
@override function bar2(): void { };
```
but it should be
```javascript
@override
function bar2(): void { };
```

File: rules/explicitOverrideRule.ts (hunk 2 above)
Comment (mzyil) [reply rules/explicitOverrideRule.ts]: fixed

File: rules/explicitOverrideRule.ts (hunk 3 above)
Comment (hmil) [reply rules/explicitOverrideRule.ts:193]: I see

**Comment by hmil** [comment]
I personally find `OPTION_NEW_LINE_AFTER_DECORATORS_AND_TAGS` a bit too verbose, but whatever. Whichever you chose, just stick to it for all constant and variable names so it's consistent.

File: rules/explicitOverrideRule.ts (hunk 3 above)
Comment (mzyil) [reply rules/explicitOverrideRule.ts:193]: So the todo stays, it doesn't seem like an urgent issue though

**Comment by mzyil** [comment]
@hmil @thehcma I think all reviews are implemented

**Comment by hmil** [comment]
Cool. Thanks for your contrib @mzyil . It should be in v0.5.0 if the Travis gods are with us.

**Pull request merged**